optimize/backend/src/main/java/io/camunda/optimize/TomcatConfig.java

```diff
@@ -0,0 +1,273 @@
+            HealthRestService.READYZ_PATH,
+            LocalizationRestService.LOCALIZATION_PATH,
+            TomcatConfig.EXTERNAL_SUB_PATH,
+            OptimizeResourceConstants.REST_API_PATH,
+            OptimizeResourceConstants.STATIC_RESOURCE_PATH,
+            OptimizeResourceConstants.ACTUATOR_ENDPOINT,
+            PanelNotificationConstants.SEND_NOTIFICATION_TO_ALL_ORG_USERS_ENDPOINT,
+            RestConstants.BACKUP_ENDPOINT,
+            UIConfigurationRestService.UI_CONFIGURATION_PATH
+          });
+
+  private static final String HTTP11_NIO_PROTOCOL = "org.apache.coyote.http11.Http11Nio2Protocol";
+
+  @Autowired private ConfigurationService configurationService;
+
+  @Autowired private Environment environment;
+
+  @Bean
+  WebServerFactoryCustomizer<TomcatServletWebServerFactory> tomcatFactoryCustomizer() {
+    LOG.debug("Setting up connectors...");
+    return new WebServerFactoryCustomizer<TomcatServletWebServerFactory>() {
+      @Override
+      public void customize(final TomcatServletWebServerFactory factory) {
+        final Optional<String> contextPath = getContextPath();
+        if (contextPath.isPresent()) {
+          factory.setContextPath(contextPath.get());
+        }
+
+        // NOTE: With the current implementation, we are always installing 2 connectors,
+        //   one for HTTP, one for HTTPs. The latter can be HTTP/1.1 or HTTP/2 depending
+        //   on the configuration.
+
+        factory.addConnectorCustomizers(
+            connector -> {
+              configureHttpConnector(connector);
+            });
+
+        factory.addAdditionalTomcatConnectors(
+            new Connector() {
+              {
+                configureHttpsConnector(this);
+              }
+            });
+      }
+    };
+  }
+
+  @Bean
+  ServletContextInitializer externalResourcesServlet() {
+    LOG.debug("Registering servlet 'externalResourcesServlet'...");
+    return new ServletContextInitializer() {
+      @Override
+      public void onStartup(final ServletContext servletContext) throws ServletException {
+        LOG.debug("Registering bean externalResourcesHandler...");
+        final URL webappURL = getClass().getClassLoader().getResource("webapp");
+        if (webappURL == null) {
+          LOG.debug("Static content directory 'webapp' not found. No bean will be registered.");
+          return;
+        }
+
+        final String webappPath = webappURL.toExternalForm().replaceFirst("file:", "");
+        final ServletRegistration.Dynamic webappServlet =
+            servletContext.addServlet("external-home", ExternalHomeServlet.class);
+        webappServlet.setInitParameter("resourceBase", webappPath);
+        webappServlet.addMapping("/external/*");
+        webappServlet.addMapping("/*");
+        webappServlet.setLoadOnStartup(1);
+      }
+    };
+  }
+
+  @Bean
+  FilterRegistrationBean<ExternalApiRewriteFilter> externalApiRewriter() {
```

Comment:
Should I move all of the filters in the file `FilterBeansConfig` maybe?

Reply:
move them where? To the TomcatConfig?

Reply:
in the file FilterBeansConfig.java, maybe